sdk/core/Azure.Core/tests/RetryPolicyTestBase.cs

```diff
@@ -65,6 +65,28 @@ public async Task OnlyRetriesRetriableCodes()
             Assert.AreEqual(501, response.Status);
         }
 
+        [Test]
```

Comment:
It feels like it would be good to add test coverage of a simple client where both RetryOptions and RetryPolicyOptions are passed, to validate assumptions about precedence rules.